### DISCUSSION TITLE
Blocks can be removed from underneath players, leading to terrible deaths.

### DISCUSSION THREAD

**Discussion by MithrandirCraft** [GriefPrevention/GriefPrevention#262] [Ideas]
### What behaviour is observed:
Removing blocks from underneath players is a popular method for killing in survival
### What behaviour is expected:
No killing through this method, since I though GriefPrevention allowed for the implementation of an impermeable environment where no one gets killed by any offensive pvp method.
### Steps/models to reproduce:
Log player 1, make some sort of death trap under a 1 layer thin platform of some easy to mine block.
Log player 2, get bait teleported (tpakilled), or place yourself on top of such block.
Player 1 removes the block, and the player falls from high altitude, onto ground, into lava, or whatever.

### Spigot/Craftbukkit/Paper and GriefPrevention version:
16.7.1

**Comment by RoboMWM** [comment]
What's the issue here? Are players not supposed to be able to break blocks in such a claim?

**Comment by MithrandirCraft** [comment]
I think I described the issue quite clearly, and even how to reproduce such a situation, but the key of the issue is that __you can kill players by removing blocks underneath them, thus causing them to fall__.
Of course, this is only possible for a perpetrator if you can actually remove blocks in the crime area, it has little to do with claims really.

In such a situation, the victim will usually fall from a high place and die, or fall into lava. It's that simple for the perpetrator. And actually, if you combine this with the item merging glitch I mentioned some time ago, you've got a pretty interesting way of stealing other player's items.
I think this isn't right if someone is thinking of using GriefPrevention as a total impermeability measure against all forms of greefing, for an extremely peacefull mode.

In most cases, the victim will usually be either afk on top of a dangerous structure, called by an unassuming trap oriented tpa, or baited onto a 1 layer thin floor, when the perpetrator does his evil deed. We all know many players are naïve enough to fall for these obvious traps.

I comment on this since there are already quite a few anti-pvp measures such as potion damage ineffectiveness and proximity lava, water and fire placing nullified against other players, and I think this would be a great addition to the measures. (I might be reporting some more methods I found people killing each other which could be avoided with some checks.)

**Comment by RoboMWM** [comment]
> I think I described the issue quite clearly, and even how to reproduce such a situation, but the key of the issue is that you can kill players by removing blocks underneath them, thus causing them to fall.

> We all know many players are naïve enough to fall for these obvious traps.

No, I really don't know how you run your server, sorry. Kinda sounds like you should disable all damage **imo** from the type of server it _seems_ you're trying to run. But idk since I can only assume that.

But anyways, I see potential ways to cause grief with such features enabled, such as preventing others from mining/breaking blocks when around them potentially. I guess this could be added though, that and/or a warning on tpa I suppose.

**Comment by MithrandirCraft** [comment]
@RoboMWM The tpa already comes with a default warning message I even translated to my own language xD, it's not enough for these idiots ahahah

In any case, it's an anti grief server full of kids, and they love it that way. The second someone gets hurt, there's massive crying.

**Comment by RoboMWM** [comment]
So I might want to keep this as just an addon (which [I can create for you](https://www.spigotmc.org/threads/requests-for-robomwm.293977/) if you desire) since it is likely expensive(?) to do an entity lookup for each and every block broken (and depending on how you do it, you may also need to check if player is above the block, not just beside it. Though then again, a player could attempt to place lava on top/side and then break block to allow it to flow so ¯\\\_(ツ)_/¯ )

While fire and lava placements also do a lookup, they don't occur as often as block breaking does.

**Comment by MithrandirCraft** [comment]
I understand. If it's that resource ineficient, I guess it's better left alone. But if you can think of anything more eficient in the future, please consider it. Your plugin always seemed like a protectein players/entities/environment provider, since it's greefing restrictions are so strong. So preventing things like these can make a great step towards it's perfection.

**Comment by RoboMWM** [comment]
Kinda depends on how many players you have, so if you don't have that many it might not be that significant. You don't know until you try.
